lib/Doctrine/ORM/Query/Filter/DefaultFilterFactory.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\ORM\Query\Filter;
+
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\EntityManagerInterface;
+
+class DefaultFilterFactory implements FilterFactoryInterface
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+
+    /**
+     * @var Configuration
+     */
+    private $ormConfiguration;
+
+    public function __construct(Configuration $ormConfiguration)
+    {
+        $this->ormConfiguration = $ormConfiguration;
+    }
+
+    public function setEntityManager(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createFromName($name)
+    {
+        $filterClass = $this->ormConfiguration->getFilterClassName($name);
+        return new $filterClass($this->entityManager);
```

Comment:
I can't see how you'd be able to use constructor injection without running into circular dependency issues since the `FilterFactory` is part of the `Configuration` which is provided to the EM.

I suppose you could take the same approach as is used for changing the `ClassMetadataFactory` - passing the class name into the EM rather than an object and having the EM instantiate it, passing itself to the `FilterFactory` constructor. I'm not a fan of this approach personally though as it means you can't have other dependencies in your factory.

Personally I think I would favour having the EM pass itself to the Factory via `createFromName` rather than having the Factory be permanently tied to a specific EM. You might want to share the same Factory for multiple EMs for example. If you want to create a factory for each EM rather than sharing them, you'd just set up your DI container to generate the EM `Configuration` objects accordingly.